plugins/jade/src/main/scala/org/broadinstitute/monster/sbt/ClassGenerator.scala

```diff
@@ -0,0 +1,126 @@
+package org.broadinstitute.monster.sbt
+
+import io.circe.jawn.JawnParser
+import org.broadinstitute.monster.sbt.model.{
+  JadeIdentifier,
+  MonsterTable,
+  SimpleColumn,
+  Struct,
+  StructColumn
+}
+
+/** Utilities for generating Scala classes from Monster table definitions. */
+object ClassGenerator {
+  private val parser = new JawnParser()
+
+  /** Scala keywords which must be escaped if used as a column / table name. */
+  private val keywords = Set("type")
+
+  /**
+    * Generate a Scala case class corresponding to a Jade table.
+    *
+    * @param tablePackage package which should contain the class
+    * @param structPackage package where any structs referenced by the table
+    *                      should be located
+    * @param tableContent JSON content of the Jade table file
+    */
+  def generateTableClass(
+    tablePackage: String,
+    structPackage: String,
+    tableContent: String
+  ): Either[Throwable, String] =
+    parser.decode[MonsterTable](tableContent).map { baseTable =>
+      val name = snakeToCamel(baseTable.name, titleCase = true)
+      val simpleFields = baseTable.columns.map(fieldForColumn)
+      val structFields = baseTable.structColumns.map(fieldForStruct(structPackage, _))
+      val classParams = (simpleFields ++ structFields).map(f => s"\n$f").mkString(",")
+
+      s"""package $tablePackage
```

Comment:
Okay this file finally makes sense to me (I was putting it off) wow this is v cool